lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
I'm not sure how to test for this.  In an rspec unit test I have: 
```Ruby
let (:facts) { {:operatingsystem => :freebsd} }
it "should work" do
  puts provider_class.default?
end
```
Which prints false every time.  How do I actually test the logic here so that as I convert the confine to a block, I'm able to ensure the same enable conditions?